library/src/include/definitions.h

```diff
@@ -14,6 +14,18 @@
  * thereby it can include top-level definitions included by all
  ******************************************************************************/
 
+// half vectors
+typedef __fp16 half8 __attribute__((ext_vector_type(8)));
+typedef __fp16 half2 __attribute__((ext_vector_type(2)));
+extern "C" half2 __v_pk_fma_f16(half2, half2, half2) __asm("llvm.fma.v2f16");
```

Comment:
There's no need to call the function __v_pk_fma_f16, although the final result is likely the v_pk_fma_f16 ISA instruction.  I'd suggest calling it llvm_fma_v2f16 since what is really being requested is the LLVM fma intrinsic on type v2f16 == half2.

Reply:
Does it matter to change the name? Using __v_pk_fma_f16 is better readable as it says it is an intrinsic and it generates v_pk_fma_f16 isa.

Reply:
It *may* generate that ISA.  But the specific request is for the LLVM intrinsic.  This change is an improvement.